infrastructure/src/main/java/org/corfudb/infrastructure/orchestrator/actions/RestoreRedundancyMergeSegments.java

```diff
@@ -58,6 +65,9 @@
     @NonNull
     private final RedundancyCalculator redundancyCalculator;
 
+    @Getter
+    private final int committedProcessorNumInFlightMessages = 25;
```

Comment:
@WenbinZhu 
I am not sure about the number. The thing is we are transferring all the segments at once, so within one transfer we can transfer a segment that has 2 source nodes and another segment that has only 1. So this number ideally should be dynamically configured but even if its the case, it is still unclear what to use for one node. I think we should settle on some number and then see how it affects the performance before introducing any dynamic configurations. I am not sure about any other approach than to eyeball it.

Reply:
@WenbinZhu I think ideally this number should not even be configured. The consumer (log unit server) must have a backpressure mechanism in place that would balance the incoming traffic. There are frameworks that achieve this easily, for example, RxJava, we don't use any of these though.

Reply:
@PavelZaytsev This number is for parallel transfer only, I think there can be a rough calculation on this number, for a layout with 2 available source log units, under the random algorithem, 25 in flight messages means every log unit will have approximately 12.5 requests at the same time, which seems too many considering we have normal workloads other than state transfer. And since the number of source log units is known, it can be configured based on it, but the problem you have now, is that every batch has it's own number of available log units, which is actually the same for parallel workloads. Ideally only the prallel batch processor should know the number of available processors, and this number can vary on every retry where layout is invalidated.

Reply:
@WenbinZhu Ok I can configure that. But what do you suggest we use as the number of in-flight requests for one source log unit server? Is 5 enough?

Reply:
Yea, we can start with 5, and later if performance is good, we can increase to somthing like 10 mabye.